plugins/hls-cabal-fmt-plugin/src/Ide/Plugin/CabalFmt.hs

```diff
@@ -0,0 +1,66 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Ide.Plugin.CabalFmt where
+
+import Control.Lens
+import Control.Monad.IO.Class
+import qualified Data.Text as T
+import Development.IDE hiding (pluginHandlers)
+import Ide.PluginUtils
+import Ide.Types
+import Language.LSP.Types as J
+import qualified Language.LSP.Types.Lens as J
+import System.Exit
+import System.FilePath
+import System.Process
+import Prelude hiding (log)
+import System.Directory
+
+data Log
+  = LogProcessInvocationFailure Int
+  | LogReadCreateProcessError String [String]
+  deriving (Show)
+
+instance Pretty Log where
+  pretty = \case
+    LogProcessInvocationFailure code -> pretty ("Invocation of cabal-fmt failed with code" :: String) <+> pretty code
+    LogReadCreateProcessError errorMsg args -> pretty ("Invocation of cabal-fmt" :: String) <+> pretty errorMsg <+> pretty ("with arguments" :: String) <+> pretty args
+
+descriptor :: Recorder (WithPriority Log) -> PluginId -> PluginDescriptor IdeState
+descriptor recorder plId =
+  (defaultCabalPluginDescriptor plId)
+    { pluginHandlers = mkFormattingHandlers (provider recorder)
+    }
+
+-- | Formatter provider of cabal fmt.
+-- Formats the given source in either a given Range or the whole Document.
+-- If the provider fails an error is returned that can be displayed to the user.
+provider :: Recorder (WithPriority Log) -> FormattingHandler IdeState
+provider recorder _ide _typ contents nfp opts = liftIO $ do
+  let cabalFmtArgs = [fp, "--indent", show tabularSize] ++ ["--tabular" | tabs]
+  x <- findExecutable "cabal-fmt"
```

Comment:
Maybe we want this to be overriable via settings?

Reply:
If we're going to do this, maybe look at how `fourmolu` does it? That plugin has a mode to use the CLI version of the tool, maybe this plugin should also have a mode option? Or I guess we can add that when/if we're able to use the library directly...

Reply:
We are not able to and Oleg is so far unresponsive to the PR https://github.com/phadej/cabal-fmt/pull/33

Additionally, cabal-fmt supports only one Cabal version, so anything else would be very difficult anyway.

Reply:
Should users install cabal-fmt manually?